ims.yaml

```diff
@@ -0,0 +1,672 @@
+          - ...tbc...
+      formattedName:
+        $ref: '#/definitions/LocalizedStringResource'
+    required:
+      - formattedNameType
+      - formattedName
+    additionalProperties: false
+  GUIDRefResource:
+    description: The data-type for establishing a Globally Unique Identifier (GUID). This is the data-type used for a 'sourcedId'. There is no predefined structure for the GUID.
+    type: object
+    properties:
+      href:
+        description: Model Primitive Datatype = AnyURI. The URI for the type of object being referenced.
+        type: string
+        format: uri
+      sourcedId:
+        description: Model Primitive Datatype = String. The globally unique identifier of the object being referenced.
+        type: string
+      type:
+        description: The type of object being referenced. This is an enumerated vocabulary.
+        type: string
+        enum:
+          - academicSession
+          - education
+          - educationOffering
+          - location
+          - organization
+          - person
+    required:
+      - href
+      - sourcedId
+      - type
+    additionalProperties: false
+  LanguageStringResource:
+    description: ...tbd (Class description)...
+    type: object
+    properties:
+      language:
+        description: Model Primitive Datatype = NormalizedString. ...tbd (PrimitiveType description)...
+        type: string
+      string:
+        description: Model Primitive Datatype = String. ...tbd (PrimitiveType description)...
+        type: string
+    required:
+      - language
+      - string
+    additionalProperties: false
+  LocalizedStringResource:
+    description: ...tbd (Class description)...
+    type: object
+    properties:
+      languageString:
+        description: ...tbd (ComplexType property description)...
+        type: array
+        minItems: 1
+        items:
+          $ref: '#/definitions/LanguageStringResource'
+    required:
+      - languageString
+    additionalProperties: false
+  MetadataResource:
+    description: This is the container for the data model extensions that may be assigned to the associated class of objects.
+    type: object
+    properties: {}
+    additionalProperties: true
+  NameResource:
+    description: ...tbd (Class description)...
+    type: object
+    properties:
+      nameType:
+        description: ...tbd (Enumeration description)...
+        type: string
+        enum:
+          - FULL
+          - FORMER
+          - PREFERRED
+          - ...tbc...
+      language:
+        description: Model Primitive Datatype = NormalizedString. ...tbd (DeriveResource property description)...
+        type: string
+      namePart:
+        description: ...tbd (ComplexType property description)...
+        type: array
+        minItems: 1
+        items:
+          $ref: '#/definitions/NamePartResource'
+    required:
+      - nameType
+      - language
+      - namePart
+    additionalProperties: false
+  NamePartResource:
```

Comment:
For all of these definitions that appear as part of the Person resource, should they be renamed to something like `NamePartSchema` or `NamePartDefinition` or something so they aren't confused with jsonapi resources? I think I just left mine as `*DType` but I should probably change them if you change the ones here. What are your thoughts?

Reply:
I agree it is good to only name jsonapi resources with *Resource. For consistency I changed these back to DType